src/main/java/uk/gov/companieshouse/api/accounts/transaction/TransactionManagerImpl.java

```diff
@@ -1,42 +1,176 @@
+
+        patchTransaction(getPatchUrl(transactionId), requestEntity);
+    }
+
+    /**
+     * Creates the rest template when class first loads
+     *
+     * @return Returns a statically created rest template
+     */
+    private static RestTemplate createRestTemplate() {
+        HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+        return new RestTemplate(requestFactory);
+    }
+
+    /**
+     * Gets the api key from the properties file
+     *
+     * @return Returns the api key if set, otherwise throw an exception
+     */
+    private String getApiKey() {
+        String apiKey = configuration.getApiKey();
+        if (apiKey.isEmpty()) {
+            throw new IllegalArgumentException("API Key property has not been set");
+        }
+        return apiKey;
+    }
+
+    /**
+     * Get the base request URL from the properties file
+     *
+     * @param transactionId - transaction id
+     * @return Get the base url if set, otherwise throw an exception
+     */
+    private String getBaseUrl(String transactionId) {
+        String url = configuration.getBaseUrl().replace(ID_PARAMETER, transactionId);
+        if (url.isEmpty()) {
+            throw new IllegalArgumentException("BaseURL property has not been set");
+        }
+        return url;
+    }
+
+    /**
+     * Get the PATCH request URL from the properties file
+     *
+     * @param transactionId - transaction id
+     * @return Get the patch url if set, otherwise throw an exception
+     */
+    private String getPatchUrl(String transactionId) {
+        String url = configuration.getPatchUrl().replace(ID_PARAMETER, transactionId);
+        if (url.isEmpty()) {
+            throw new IllegalArgumentException("PatchURL property has not been set");
+        }
+        return url;
+    }
+
+    /**
+     * Get the root uri from the properties file
+     *
+     * @return Get the root uri if set, otherwise throw an exception
+     */
+    private String getRootUri() {
+        String url = configuration.getRootUri();
+        if (url.isEmpty()) {
+            throw new IllegalArgumentException("RootURL property has not been set");
+        }
+        return url;
+    }
+
+    /**
+     * GET the transaction resource
+     *
+     * @param url - url to send the get request
+     * @param requestEntity - the request entity object
+     */
+    private ResponseEntity<Transaction> getTransaction(String url, HttpEntity requestEntity) {
+        return restTemplate.exchange(getRootUri() + url, HttpMethod.GET, requestEntity, Transaction.class);
+    }
+
+    /**
+     * Patch the transaction resource
+     *
+     * @param url - url to send the get request
+     * @param requestEntity - the request entity object
+     */
+    private void patchTransaction(String url, HttpEntity requestEntity) {
+        restTemplate.exchange(getRootUri() + url, HttpMethod.PATCH, requestEntity, Void.class);
```

Comment:
You are not checking if the patch was successful.

Reply:
How would we check if the patch was succesfull?

Reply:
What would be in the `ResponseEntity` type? 
`ReponseEntity<Transaction> resp = restTemplate.exchange(getRootUri() + url, HttpMethod.PATCH, requestEntity, Void.class);`

Reply:
It would be ResponseEntity<Void> but I believe you can use ResponseEntity<?> as you don't care about the type.

Reply:
I've inserted the logic for the checking of the status code, but the actual error to be throw will be covered by the api error story